cpp/src/sort.cpp

```diff
@@ -300,28 +319,32 @@ class SortTask : public Task<SortTask, OpCode::Sort> {
     const auto stable          = argument::get_next_scalar<bool>(ctx);
     auto output                = argument::get_next_output<PhysicalTable>(ctx);
 
-    if (tbl.is_broadcasted() && ctx.rank != 1) {
-      // Note: It might be nice to just sort locally and keep it broadcast.
-      output.bind_empty_data();
-      return;
-    }
-
     // Create a new locally sorted table (we always need this)
     auto cudf_tbl  = tbl.table_view();
     auto key       = cudf_tbl.select(keys_idx);
     auto sort_func = stable ? cudf::stable_sort_by_key : cudf::sort_by_key;
     auto my_sorted_tbl =
       sort_func(cudf_tbl, key, column_order, null_precedence, ctx.stream(), ctx.mr());
 
-    if (ctx.nranks == 1 || tbl.is_broadcasted()) {
+    if (ctx.nranks == 1) {
```

Comment:
The broadcast check needs fixing, it isn't reliable enough.  I think here the `nrank` should be sufficient, though.  (in joins this is probably a latent problem)